### ISSUE TITLE
Not preserving XHTML

### ISSUE THREAD

**Issue by Saltallica** [jonkemp/gulp-inline-css#22]
I'm attempting to use this to inline styles in an XSL template. In order for the XSLT processor to operate correctly, it requires strict XHTML with proper closing tags. For example:

```html
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
```

Becomes

```html
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
```
Without the closing tag. This is not legal XML, therefore fails processing.

Is there a way to output as XHTML with closing tags **everywhere**?

**Comment by jonkemp** [comment]
No, not currently. It would require significant changes to the codebase.

**Comment by yn5** [comment]
@jonkemp Could you point out roughly what would have to change to prevent this library from invalidates our XML so we can come up with a pull request?

**Comment by jonkemp** [comment]
@yn5 well, this plugin doesn't actually handle transforming the output. It goes through another module called inline-css.

https://github.com/jonkemp/inline-css

This module uses cheerio to do the html parsing and cheerio uses another module called htmlparser2. If the issue is the XML being mangled then I would look there. htmlparser2 says it can handle XML.

**Comment by mark-making** [comment]
Anyone coming here for a solution, just pass `xmlMode: true` with the other options:

```
.pipe(inline({ preserveMediaQueries: true, removeStyleTags: false, removeLinkTags: true, xmlMode: true }))
```

**Comment by BroFox86** [comment]
I personally use [gulp-htmltidy](https://www.npmjs.com/package/gulp-htmltidy) to convert HTML to XHTML again with this doctype option:

```
export function build() {
  return src("temp/*.html")
    .pipe(plugins.inlineCss())
    .pipe(plugins.htmltidy({
      doctype: "xhtml",
      hideComments: false,
      indent: false
    }))
    .pipe(plugins.htmlBeautify())
    .pipe(dest("dist/"));
}
```
